src/main/webapp/resources/js/pages/projects/samples/components/MergeModal.jsx

```diff
@@ -0,0 +1,182 @@
+import {
+  Alert,
+  Checkbox,
+  Col,
+  Form,
+  Input,
+  Modal,
+  notification,
+  Radio,
+  Row,
+  Space,
+} from "antd";
+import React from "react";
+import { useSelector } from "react-redux";
+import { serverValidateSampleName } from "../../../../utilities/validation-utilities";
+import { useMergeMutation } from "../../../../apis/projects/samples";
+import LockedSamplesList from "./LockedSamplesList";
+
+/**
+ * React element to display a modal to merge multiple samples into a single one.
+ * @param {array} samples - list of samples to merge together
+ * @param {boolean} visible - whether the modal is currently visible on the page
+ * @param {function} onComplete - function to call when the merge is complete
+ * @param {function} onCancel - function to call when the merge is cancelled.
+ * @returns {JSX.Element}
+ * @constructor
+ */
+export default function MergeModal({ samples, visible, onComplete, onCancel }) {
+  const { projectId } = useSelector((state) => state.samples);
+  const [merge, { isLoading }] = useMergeMutation();
```

Comment:
I don't think I've seen this before. Where is it getting the `isLoading` from? We usually get it from the state

Reply:
https://redux-toolkit.js.org/rtk-query/usage/mutations#frequently-used-mutation-hook-return-values